lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -342,12 +342,11 @@ def gem_version_compare(first, second)
                      curl --cacert $SSL_CA_CERT https://smart-proxy.example.com:8443"
           end
           def foreman_server_ca_cert
-            if File.exist?(Setting[:ssl_ca_file])
-              File.read(Setting[:ssl_ca_file])
-            else
-              msg = N_("SSL CA file not found, check the 'SSL CA file' in Settings > Authentication")
-              raise Foreman::Exception.new(msg)
-            end
+            raise N_("No CA file set, check the 'SSL CA file' in Settings > Authentication") if Setting[:ssl_ca_file].blank?
+            File.read(Setting[:ssl_ca_file])
+          rescue => e
+            msg = N_("%s, check the 'SSL CA file' in Settings > Authentication") % e.message
+            raise Foreman::Exception.new(msg)
```

Comment:
I think we should add and use a specific exception for this. That way it gets its own code and we can provide a documentation for such state. See https://github.com/theforeman/foreman/blob/develop/lib/foreman/renderer/errors.rb#L10-L44

Or we could use `render_error`, but I think the first is better.

Reply:
Just to clarify: That exception should be for empty setting, not for missing file, right?
Something like:
```
      class EmptySetting < RenderingError
        MESSAGE = N_("message").freeze
      end
```
Suggestions for class name & error message are more than welcomed!